src/subdomains/staking/application/services/staking-deposit.service.ts

```diff
@@ -195,9 +197,11 @@ export class StakingDepositService {
         }
 
         // verify first pay in
-        const payInValid = staking.getConfirmedDeposits().length > 0 || (await this.isFirstPayInValid(staking, payIn));
+        const payInValid =
+          (await this.depositRepository.getConfirmed(stakingId)).length > 0 ||
+          (await this.isFirstPayInValid(staking, payIn));
         if (payInValid) {
-          this.createOrUpdateDeposit(staking, payIn);
+          await this.createOrUpdateDeposit(staking, payIn);
         } else {
           console.error(`Invalid first pay in, staking ${staking.id} is blocked`);
           staking.block();
```

Comment:
Staking save (from below) can be moved in here?

Reply:
great catch! previously we needed to save in any case because we were pushing the new deposit to staking, now we can save only on block